workshop/posting-error-example.md

```diff
@@ -0,0 +1,201 @@
+       - What kind of file *e.g.* `.TSV`, `.fastq`?
+
+- Sometimes you can bypass an error completely by using an alternative strategy altogether, but your helper won't be able to suggest an alternative if they don't know your ultimate goal.
+
+*For our example:*  
+<img src = "screenshots/what-is-the-goal.png" width = "600">
+
+### What is the code that is producing this error?
+
+- You should [use backticks (```) to format your code.](../virtual-setup/slack-procedures.md#adding-code-blocks-to-messages) (How to do this in the next step).
+Formatting your code makes it easier for others to distinguish between your code and the words in your post where you are otherwise describing the problem.
+
+- Screenshots are not ideal because others can't copy-paste your code from a screenshot. Being able to copy-paste the code with backticks makes it easiest for others to help you.
+
+*For our example:*  
+<img src = "screenshots/what-is-the-code.png" width = "600">
+
+### What is the error or problematic outcome?  
+
+- Include any error messages verbatim. Also preferably not a screenshot so others can copy-paste for Google searching.
+- If you don't have an error message per se, but do have something else that is not working as expected, describe what isn't working and how you found it.
+     - *Example*: "This data.frame I'm showing in my code, called `cool_df` doesn't have row names when I look at it in my environment panel".
+- Context around when the error appears versus when it does not can be helpful clues to your reader.
+
+*For our example:*  
+<img src = "screenshots/what-is-the-error.png" width = "600">
+
+### (Optional) What have you tried thus far?  
+
+- This is considered an optional point because we understand you also may not know where to begin (and that is definitely okay - we are here to help!)
+
+- If you have tried some things to fix the problem; it can be helpful for others to know what you've tried and narrow down on what the problem might be.
+
+- If you have tried solutions you have found posted online, including the links to those solutions is helpful.
+
+*For our example:*  
+<img src = "screenshots/what-have-you-tried.png" width = "600">
+
+## Step 3) Post to Slack!
+
+Navigate to your particular workshop's training channel.
+
+If you wrote out our example question post in a text editor, it may have looked like this; where backticks will become code chunks after we copy-paste it in Slack:
+
+<img src = "screenshots/text-editor.png" width = "500">
+
+When you are typing out your question you may find it helpful to have our "four points" as your headers but that is up to you.
+We've also included [a checklist you could use for creating your post](#posting-a-code-question-checklist).
+
+When we copy and paste our text in Slack, it may ask  you if you'd like to `Apply formatting`. Click `Apply`:
+
+<img src = "screenshots/slack-apply-formatting.png" width = "500">
+
+This will change your post to be formatted:
```

Comment:
What if it doesn't? I ask because I have certainly had pasted code not get formatted, for unknown reasons. We should probably encourage them to look to see if the formatting is what they expect, and to add code formatting (& the difference between `code` and `code block`?) if Slack messes up (or they don't put all the formatting in their text). 

I don't think you have anywhere that you tell people to give variable names in their text `code format`, but that is also helpful... (not as important as the main code block though)

Reply:
Add the `Shift Command F` tip for applying formatting. I'm worried about also asking for variables to be in backticks. This document is already prescribing a lot.